python/tvm/driver/tvmc/compiler.py

```diff
@@ -40,17 +40,38 @@
 from .transform import convert_graph_layout
 from .shape_parser import parse_shape_string
 from .workspace_pools import generate_workspace_pools_args, workspace_pools_recombobulate
+from .extensions import load_extensions, get_extensions
+from .arguments import TVMCSuppressedArgumentParser
 
 # pylint: disable=invalid-name
 logger = logging.getLogger("TVMC")
 
 
 @register_parser
-def add_compile_parser(subparsers, _, json_params):
+def add_compile_parser(subparsers, main_parser, json_params, argv):
     """Include parser for 'compile' subcommand"""
 
-    parser = subparsers.add_parser("compile", help="compile a model.")
+    parser = subparsers.add_parser("compile", help="compile a model.", add_help=False)
```

Comment:
Curious, why do we need to remove help option?

Reply:
I encountered an issue where the TVMCSuppressedArgumentParser still caused a SystemExit exception. However, I cannot seem to reproduce the issue right now. So I'll remove this again.

Reply:
Alright, I found the input again: "tvmc compile --help". When removing the help option from the subparser, this is resolved. The help option still works because it is enabled through the main parser.

Reply:
I tried this out locally but see some difference in the help prompt when executing `tvmc compile --help`. Before the patch we get something like:
```
$ tvmc compile --help
usage: tvmc compile [-h] [--cross-compiler CROSS_COMPILER]
                    [--cross-compiler-options CROSS_COMPILER_OPTIONS]
                    [--desired-layout {NCHW,NHWC}] [--dump-code FORMAT]
                    [--model-format {keras,onnx,pb,tflite,pytorch,paddle,relay}]
                    [-o OUTPUT] [-f {so,mlf}] [--pass-config name=value]
                    [--target TARGET]
...
positional arguments:
  FILE                  path to the input model file.

optional arguments:
  -h, --help            show this help message and exit
  --cross-compiler CROSS_COMPILER
                        the cross compiler to generate target libraries, e.g.
                        'aarch64-linux-gnu-gcc'.
  --cross-compiler-options CROSS_COMPILER_OPTIONS
                        the cross compiler options to generate target
                        libraries, e.g. '-mfpu=neon-vfpv4'.
  --desired-layout {NCHW,NHWC}
                        change the data layout of the whole graph.
```
While after seems to lead to:
```
$ tvmc compile --help
usage: tvmc compile [--experimental-tvm-extension EXPERIMENTAL_TVM_EXTENSION]
                    [--cross-compiler CROSS_COMPILER]
                    [--cross-compiler-options CROSS_COMPILER_OPTIONS]
                    [--desired-layout {NCHW,NHWC}] [--dump-code FORMAT]
                    [--model-format {keras,onnx,pb,tflite,pytorch,paddle,relay}]
                    [-o OUTPUT] [-f {so,mlf}] [--pass-config name=value]
                    [--target TARGET]
...
tvmc compile: error: the following arguments are required: FILE
```
I think it would be really helpful if we can keep the argument descriptions as before

Reply:
Alright, I can confirm this issue. The underlying issue is that the `TVMCSuppressedArgumentParser` still throws `SystemExit` when given such a command line. We also cannot override `error` of `ArgumentParser`, because the failing parser is the subparser, which is not a `TVMCSuppressedArgumentParser`.

Before this patch, this problem already occurs. For example with `tvmc run --help`, `SystemExit` is thrown and a partial help message is printed (missing `--project-option`). So my suggested fix is to move the `parse_known_args` as far down as possible to yield a mostly complete help message.

Reply:
Thanks for investigating! This looks much better, so I won't block on it. Perhaps we can create a separate issue to track this bug?